contracts/Distributor.sol

```diff
@@ -0,0 +1,259 @@
+        IERC20 token = token_;
+        uint totalPayees = numRegisteredForToken(token);
+        for(uint256 i = 0; i < totalPayees;){
+            payoutToken(_payeesForToken[token][i], token);
+            unchecked {i++; }
+        }
+    }
+
+      /*///////////////////////////////////////////////////////////////
+                            Internal/Private Functions
+    //////////////////////////////////////////////////////////////*/
+
+    // @Desc: To register accounts as payees for token distribution.
+    // @Note: tokenAddress_ to specify as Null Address to register a payee with shares for Ether.
+    function _registerPayeesForETH(address account_, uint256 share_) private nonZeroAddress(account_) validShares(share_) {
+        address account = account_;
+        uint share = share_;
+
+        require(_sharesForETH[account_] == 0, "Distributor: Account registered for ETH distribution.");
+
+        // Save account as a payee in the mapping
+        _payeesForETH[_numRegisteredForETH++] = account_; 
+        _sharesForETH[account_] = share_;
+         _totalETHShares += share_;
+         
+        emit RegisterPayeeForETH(account, share);
+    }
+
+    function _registerPayeesForTokens(address account_, IERC20 token_, uint256 share_) private nonZeroAddress(account_) validShares(share_) {
+        address account = account_;
+        uint256 share = share_;
+        IERC20 token = token_;
+
+        require(_sharesForTokens[token][account] == 0, "Distributor: Account already registered for token distribution,");
+
+        _payeesForToken[token][_numRegisteredForToken[token]] = account;
+        _numRegisteredForToken[token] += 1;
+      
+
+        _sharesForTokens[token][account] = share;
+        _totalTokenShares[token] += share;
+
+        emit RegisterPayeeForToken(account, token, share);
+    }
+
+    // @Desc: To calculate the outstanding amount for an account to claim from the contract based on how much total ether this contract received since its inception
+    function outstandingETHClaim(address account_, uint256 totalReceived_, uint256 accountDistributed_) private view returns(uint256){
+        return (totalReceived_ * _sharesForETH[account_]) / totalETHShares() - accountDistributed_;
+    }
+
+    // @Desc: To calculate outstanding amount of an ERC20 for an account to claim from the contract based on how much it has received since inception
+    function outstandingTokenClaim(address account_, IERC20 tokenAddress_, uint totalTokenReceived_, uint256 accountTokenDistributed_) private view returns(uint256){
+        return (totalTokenReceived_ * _sharesForTokens[tokenAddress_][account_]) / totalTokenShares(tokenAddress_) - accountTokenDistributed_;
+    }
+
+    
+      /*///////////////////////////////////////////////////////////////
+                            Pure/View Functions
+    //////////////////////////////////////////////////////////////*/
+    function numRegisteredForETH() public view returns(uint256){
```

Comment:
Is there a reason for you to make the variable private & write public functions?